### PR TITLE
If ledger not enabled, honor settings for clearing history

### LINKED ISSUE #8495: If ledger not enabled, honor settings for clearing history
- Did you search for similar issues before submitting this one? yes

- Describe the issue you encountered: if the ledger isn't enabled, then the user's settings for clearing history should be honored

- Platform (Win7, 8, 10? macOS? Linux distro?): all

- Brave Version (revision SHA): n/a

- Steps to reproduce:
    1. do *not* enable the ledger
    2. visit various sites
    3. go to about:preferences#security and under "Private Data" enable "Browsing history" to be cleared and then quit the browser
    4. *or* go to about:preferences#security and click "Clear Browsing Data Now...", enable "Browsing history" to be cleared
    
- Actual result: ledger-synopsis files are not removed

- Expected result: ledger-files files are not removed

- Will the steps above reproduce in a fresh profile? If not what other info can be added? no, you need to visit some 

- Is this an issue in the currently released version? yes

- Can this issue be consistently reproduced? yes

- Extra QA steps: none
 
- Screenshot if needed: none

- Any related issues: #6708 deals with moving ledger-specific settings for sites from site settings to the `ledger-synopsis` file

### PR DESCRIPTION
**Pull request by mrose17**
Fixes #8495

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Test Plan:

### CODE REVIEW

**Comment by bsclifton** [comment]
Pulling into 0.15.0 since this can speed up load time significantly

**Comment by NejcZdovc** [comment]
@bsclifton will add test for it

**Comment by mrose17** [comment]
@NejcZdovc - thank you!

**Comment by NejcZdovc** [comment]
let's wait for travis but I tested all 3 tests locally and they are green. @bsclifton can you please do a final review

File: app/ledger.js [hunk 1]
```diff
@@ -263,19 +269,23 @@ const doAction = (action) => {
...
  * module entry points
  */
+
 var init = () => {
   try {
     appDispatcher.register(doAction)
     initialize(getSetting(settings.PAYMENTS_ENABLED))
 
-    doneTimer = setInterval(doneWriter, 1 * msecs.hour)
+    doneTimer = setInterval(doneWriter, 1 * msecs.minute)
```
Comment (bsclifton) [inline app/ledger.js]: Was this intentionally bumped down to 1 minute?

Comment (bsclifton) [inline app/ledger.js]: (I talked with @mrose17; it's fine as original or as 1 minute, but he advised 1 minute is optimal)

**Pull request merged**